x-pack/plugins/cases/public/client/helpers/group_alerts_by_rule.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { CommentRequestAlertType } from '../../../common/api';
+import { CommentType, Ecs } from '../../../common';
+import { getRuleIdFromEvent } from './get_rule_id_from_event';
+import { CaseAttachments } from '../../types';
+
+type Maybe<T> = T | null;
+interface Event {
+  data: EventNonEcsData[];
+  ecs: Ecs;
+}
+interface EventNonEcsData {
+  field: string;
+  value?: Maybe<string[]>;
+}
+
+export const groupAlertsByRule = (items: Event[], owner: string): CaseAttachments => {
+  const attachmentsByRule = items.reduce<Record<string, CommentRequestAlertType>>((acc, item) => {
+    const rule = getRuleIdFromEvent(item);
+    if (!acc[rule.id]) {
+      acc[rule.id] = {
+        alertId: [],
+        index: [],
+        owner,
+        type: CommentType.alert as const,
+        rule,
+      };
+    }
+    const alerts = acc[rule.id].alertId;
+    const indexes = acc[rule.id].index;
+    if (Array.isArray(alerts) && Array.isArray(indexes)) {
+      alerts.push(item.ecs._id ?? '');
+      indexes.push(item.ecs._index ?? '');
+    }
+    return acc;
+  }, {});
+  const attachments: CaseAttachments = [];
+  const keys = Object.keys(attachmentsByRule);
+  for (const key of keys) {
+    const x = attachmentsByRule[key];
+    attachments.push(x);
+  }
+  return attachments;
```

Comment:
`return Object.values(attachmentsByRule);`?

Reply:
I always keep forgetting about Object.values. mostly because I keep thinking it uses iterators and we can't use iterators in kibana. Thanks for pointing it!